.github/workflows/code-checks.yml

```diff
@@ -94,11 +82,9 @@ jobs:
 
     - name: Run typing validation
       run: ci/code_checks.sh typing
-      if: ${{ steps.build.outcome == 'success' }}
```

Comment:
I think these 2 `if`s should be here

Reply:
I don't think so because both require a working `pandas` import, or am I missing something?

Reply:
In my experience these steps will still "run" and fail for the reason you brought up. i.e. it reduces the noise in the build outputs by just skipping this is the build failed

Reply:
Oh you are right we actually wanted the same thing but I confused the meaning of this option. Will change